### PR TITLE
Connect: Add theme configuration

### PR DESCRIPTION
**Pull request by gzdunek**
Contributes to https://github.com/gravitational/teleport/issues/27079

Enables Connect theme configuration. The available options are: `light`, `dark`, `system` (sync with OS).
The default is `system`.

The implementation turned to be slightly more complex than I thought, because of two Linux-related issues:
- `prefers-color-scheme` [is not updated](https://github.com/electron/electron/issues/21427#issuecomment-589796481) when the system theme changes. There [is a bug in Chromium](https://bugs.chromium.org/p/chromium/issues/detail?id=998903) related to to this. Apparently it was fixed in Chromium 114, but people are still reporting issues with that (which is interesting, I tested Chromium 114 on my Ubuntu VM and webpages seemed to respect the system theme). Probably there have be some work done on the Electron side, because I tested Electron 25 (Chromium 114) and it was not better than Electron 24. 
The workaround that I added relies on `nativeTheme` updates that work correctly. These updates are then passed to the renderer via IPC and `webContents.send`.
- On Fedora, `nativeTheme` is not updated at all 😓. It is a [known issue](https://github.com/electron/electron/issues/33635#issuecomment-1502215450). Because of that, the app is always rendered in the light mode and the user will have to switch do dark in the app settings if they won't like the light one. Theoretically, we could detect Fedora and use `dark` as a default value there, but I don't know if this is worth it, probably not.
I also took a look at vscode, but they don't have any smart fix for that.

(BTW I still have to work on the appearance of the tabs)

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

**Review by avatus — APPROVED** [review]

**Review by ravicious — APPROVED** [review]
I tested it and it works quite well, good job!

I'm not actually sure how we could use `prefers-color-scheme` here, it seems like `nativeTheme` is naturally a better fit for us here. 🤔

File: web/packages/teleterm/src/mainProcess/windowsManager.ts [hunk 1]
```diff
@@ -88,6 +98,13 @@ export class WindowsManager {
       this.popupUniversalContextMenu(window, props);
     });
 
+    nativeTheme.on('updated', () => {
+      window.webContents.send(
+        'main-process-subscribe-to-native-theme-update',
```
[the remark this answers is not in the document]
Comment (gzdunek) [reply web/packages/teleterm/src/mainProcess/windowsManager.ts:103]: Done

File: web/packages/teleterm/src/ui/ThemeProvider/ThemeProvider.tsx [hunk 2]
```diff
@@ -14,22 +14,53 @@ See the License for the specific language governing permissions and
...
+  // (for example, when the app was launched in a dark mode, it always returns 'dark'
+  // ignoring that the system theme is now 'light').
+  // Related Electron issue: https://github.com/electron/electron/issues/21427#issuecomment-589796481,
+  // Related Chromium issue: https://bugs.chromium.org/p/chromium/issues/detail?id=998903
+  //
+  // Additional issue is that nativeTheme does not return correct values at all on Fedora:
+  // https://github.com/electron/electron/issues/33635#issuecomment-1502215450
+  const ctx = useAppContext();
+  const [activeTheme, setActiveTheme] = useState(
+    ctx.mainProcessClient.getNativeTheme() === 'dark' ? darkTheme : lightTheme
```
[the remark this answers is not in the document]
Comment (gzdunek) [reply web/packages/teleterm/src/ui/ThemeProvider/ThemeProvider.tsx]: Ah good catch, I didn't think of it.

[the remark this answers is not in the document]
Comment (gzdunek) [reply web/packages/teleterm/src/ui/ThemeProvider/ThemeProvider.tsx]: Good idea, there is not much sense in converting boolean to string and then back to boolean (`ctx.mainProcessClient.getNativeTheme() === 'dark'`). I changed it also for the subscription.

File: web/.storybook/preview.js [hunk 3]
```diff
@@ -36,22 +40,26 @@ if (typeof global.process === 'undefined') {
...
-
   if (meta.title.startsWith('Teleterm/')) {
-    ThemeProvider = TeletermThemeProvider;
-    theme = TeletermTheme;
+    const theme =
+      meta.globals.theme === 'Dark Theme'
+        ? teletermDarkTheme
+        : teletermLightTheme;
+    return (
+      <MockAppContextProvider>
```
[the remark this answers is not in the document]
Comment (gzdunek) [reply web/.storybook/preview.js]: Done.

**Comment by gzdunek** [comment]
>I'm not actually sure how we could use prefers-color-scheme here, it seems like nativeTheme is naturally a better fit for us here. 🤔

If we have only light and dark theme, `prefers-color-scheme` is fine. It changes to `prefers-color-scheme: dark` when `shouldUseDarkColors` is `true`, and to `prefers-color-scheme: light` when not. We could watch for changes using `window.matchMedia` and not have to add two IPCs :)

**Pull request merged**